src/data/text_parser.h

```diff
@@ -111,7 +111,7 @@ inline bool TextParserBase<IndexType, DType>::FillData(
     std::vector<RowBlockContainer<IndexType, DType> > *data) {
   InputSplit::Blob chunk;
   if (!source_->NextChunk(&chunk)) return false;
-  const int nthread = omp_get_max_threads();
+  const int nthread = this->nthread_;
```

Comment:
This ensures that `nthread` parameter is honored by the text parser. Previously, the text parser would always use all CPU cores.

Reply:
I didn't modify this file but this looked like a bug previously introduced. In my practice, CPU usage never exceeds 500% on a 64-core CPU. 
Also, do we need to change this too? https://github.com/dmlc/dmlc-core/blob/main/src/data.cc#L59 It looks like nthreads_ has been hardcoded as 2. Otherwise, users may experience performance drop.

Reply:
I fixed this particular bug because it was breaking the unit test for the Parquet parser. Let's solve other bugs in future pull requests.